### ISSUE TITLE
slashes_and_izarc.zip difference between file name (central dir vs local file header)

### ISSUE THREAD

**Issue by Prinzhorn** [Stuk/jszip#251]
I'm writing a tool which lists all contents of a zip (without decompressing or anything, just a list of filenames). I'm using the zip files from this repo for my test suite. I've noticed something about the slashes_and_izarc.zip file and wonder if you can help me out.

The file name of the content is `test\Hello.txt`. The central directory contains `test\Hello.txt` as file name. But the local file header contains `test/Hello.txt` as file name field (notice the forward slash). This gives the impression that there's a folder. The spec says

> All slashes should be forward slashes '/' as opposed to backwards slashes '\' for compatibility with Amiga and UNIX file systems etc.

https://github.com/Stuk/jszip/blob/3a39a715a5c522686aad95026134be3d5731d06e/docs/APPNOTE.TXT#L679

So basically my question is: **is this a bug in izarc**? Which of the both entries should I trust in general? A few minutes ago I was expecting them to be always identical.

Thank you for your help!

**Comment by dduponchel** [comment]
I don't have the time right now to write a long response, but you may be interested in [this comment](https://github.com/Stuk/jszip/blob/v2.5.0/lib/zipEntry.js#L90) in the source code. We get the filename from the local file headers to avoid these '\' and everything else from the central directory.

**Comment by Prinzhorn** [comment]
Thank you very much, the short response is all I needed. Since I have control over the generated zip files and don't need to be a general purpose parser I can just assume there won't be backslashes in the file names. Or I can even reject files where both file name fields don't match.
